sort/mergesort2.go

```diff
@@ -0,0 +1,37 @@
+package sort
+
+import "github.com/TheAlgorithms/Go/math/min"
+
+func Merge2(items []int) []int {
```

Comment:
I believe `MergeIter` would be a better name for this function. `Merge2` does not convey that much information.

Reply:
In that case, same applies to `Merge` as well. Maybe a documentation comment here would be better. Not sure if you agree.

I think having a consistent way to do it would be better and we should come up with a naming convention or something for this repo.

Reply:
According to [Contribution Guidelines](https://github.com/TheAlgorithms/Go/blob/master/CONTRIBUTING.md) :
- If an implementation of the algorithm already exists and your version is different from that implemented, please use incremental numeric digit as a suffix. For example: if `binarysearch.go` already exists in the search folder, and you are contributing a new implementation, the filename should be `binarysearch2.go` and for a third implementation, `binarysearch3.go`.

Therefore,  if the exist recursive implementation of  merge sort  called `MegeRecursion` ,  `MergeIter` would be a better name for this function.

Reply:
@tjgurwara99 we could rename the existing `sorts.Merge` to `sorts.MergeRecursive`

Reply:
I agree with this. We should update the contribution guidelines accordingly as well.